static/js/compose_validate.js

```diff
@@ -520,16 +520,19 @@ export function validation_error(error_type, stream_name) {
     return true;
 }
 
-export function validate_stream_message_address_info(stream_name) {
+export function validate_stream_message_address_info(stream_name, send_anyway) {
     if (stream_data.is_subscribed_by_name(stream_name)) {
         return true;
     }
     const autosubscribe = page_params.narrow_stream !== undefined;
     const error_type = check_unsubscribed_stream_for_send(stream_name, autosubscribe);
+    if (error_type === "not-subscribed" && send_anyway) {
+        return true;
+    }
```

Comment:
Would it be cleaner to have this be a third parameter passed to `check_unsubscribed_stream_for_send`, like `autosubscribe`?

Reply:
DONE: more code, but splits-out autosubscribe as two different cases. Added a nodejs test.